src/proj7BittingCerratoCohenEllmer/bantam/parser/Parser.java

```diff
@@ -876,6 +872,12 @@ private void ensureTokenType(String errorMessage, Token.Kind... kinds) {
         }
     }
 
+    /**
+     * Helper method that registers an error with the internal error handler and throws
+     * the exception to stop the parsing.
+     *
+     * @param errorMessage to display in the error handler and in the runtime exception
+     */
     private void registerAndThrow(String errorMessage) {
         errorHandler.register(Error.Kind.PARSE_ERROR, errorMessage);
```

Comment:
Theres another errorHandler.register call that takes the filename and linenumber and formats a more useful error message. I dont think that this is a change that should hold things up.